perses/rjmc/geometry.py

```diff
@@ -198,31 +200,34 @@ def logp_reverse(self, top_proposal, new_coordinates, old_coordinates):
                 #get the internal coordinate representation
                 #we want to see the probability of old_atom where all other coordinates are new_atom
                 atom_coords = old_coordinates[atom.idx]
-                bond_coords = reverse_proposal_coordinates[bond_atom.idx]
-                angle_coords = reverse_proposal_coordinates[angle_atom.idx]
-                torsion_coords = reverse_proposal_coordinates[torsion_atom.idx]
+                bond_coords = old_coordinates[bond_atom.idx]
+                angle_coords = old_coordinates[angle_atom.idx]
+                torsion_coords = old_coordinates[torsion_atom.idx]
                 internal_coordinates, detJ = self._cartesian_to_internal(atom_coords, bond_coords, angle_coords, torsion_coords)
+                r = internal_coordinates[0]*units.angstrom
+                theta = internal_coordinates[1]*units.radian
+                phi = internal_coordinates[2]*units.radian
 
                 #propose a bond and calculate its probability
                 bond = self._get_relevant_bond(atom, bond_atom)
                 bond_k = bond.type.k
                 sigma_r = units.sqrt(1/(beta*bond_k))
                 logZ_r = np.log((np.sqrt(2*np.pi)*sigma_r/sigma_r.unit)) #need to eliminate unit to allow numpy log
-                logp_r = self._bond_logq(internal_coordinates[0], bond, beta) - logZ_r
+                logp_r = self._bond_logq(r, bond, beta) - logZ_r
 
                 #propose an angle and calculate its probability
                 angle = self._get_relevant_angle(atom, bond_atom, angle_atom)
                 angle_k = angle.type.k
                 sigma_theta = units.sqrt(1/(beta*angle_k))
                 logZ_theta = np.log((np.sqrt(2*np.pi)*sigma_theta/sigma_theta.unit)) #need to eliminate unit to allow numpy log
-                logp_theta = self._angle_logq(internal_coordinates[1], angle, beta) - logZ_theta
+                logp_theta = self._angle_logq(theta, angle, beta) - logZ_theta
 
                 #calculate torsion probability
-                logp_phi = self._torsion_logp(atom, atom_coords, torsion, atoms_with_positions, reverse_proposal_coordinates, beta)
+                logp_phi = self._torsion_logp(atom, atom_coords, torsion, old_atoms_with_positions, old_coordinates, beta)
                 logp = logp + logp_choice + logp_r + logp_theta + logp_phi + np.log(detJ)
```

Comment:
yeah, it was the bond. The length was cut by a factor of 10 as a result of a unit error.